physionet-django/user/citi_training_module.py

```diff
@@ -0,0 +1,155 @@
+        - 'strCompletionReport': The completion status of the course (e.g.,'Human Research').
+        - 'intGroupID': The ID of the course group (e.g., '43007').
+        - 'strGroup': The name of the course group (e.g., 'Data or Specimens Only Research').
+        - 'intStageID': The ID of the course stage (e.g., '106240').
+        - 'intStageNumber': The number of the course stage within the group (e.g., '1').
+        - 'strStage': The name of the course stage (e.g., 'Basic Course').
+        - 'intCompletionReportID': The ID of the completion report.
+        - 'intMemberStageID': The ID of the member stage.
+        - 'dtePassed': The date and time the course was completed, as a datetime object.
+        - 'intScore': The score earned for the course.
+        - 'intPassingScore': The passing score for the course.
+        - 'dteExpiration': The date and time the course will expire, as a datetime object.
+
+        If no courses are found for the given email address, returns an empty list.
+    """
+
+    username = settings.CITI_USERNAME
+    password = settings.CITI_PASSWORD
+
+    memberid = get_memberid(email)
+
+    if memberid is None:
+        return []
+
+    payload_courses = loader.render_to_string('user/citi/get_courses_by_member.xml', {
+        'username': username, 'password': password, 'memberid': memberid, })
+
+    root_courses = send_request(xml_payload=payload_courses)
+
+    completion_info = []
+    for crs in root_courses.findall('.//CRSMEMBERID'):
+        FirstName = crs.find('FirstName')
+        LastName = crs.find('LastName')
+        MemberID = crs.find('MemberID')
+        memberEmail = crs.find('memberEmail')
+        strCompletionReport = crs.find('strCompletionReport')
+        strGroup = crs.find('strGroup')
+        intGroupID = crs.find('intGroupID')
+        intStageID = crs.find('intStageID')
+        intStageNumber = crs.find('intStageNumber')
+        strStage = crs.find('strStage')
+        intCompletionReportID = crs.find('intCompletionReportID')
+        intMemberStageID = crs.find('intMemberStageID')
+        intScore = crs.find('intScore')
+        intPassingScore = crs.find('intPassingScore')
+        dtePassed = crs.find('dtePassed')
+        dteExpiration = crs.find('dteExpiration')
+
+        completion_info.append({
+            'FirstName': FirstName.text,
+            'LastName': LastName.text,
+            'MemberID': MemberID.text,
+            'memberEmail': memberEmail.text,
+            'strCompletionReport': strCompletionReport.text,
+            'intGroupID': intGroupID.text,
+            'strGroup': strGroup.text,
+            'intStageID': intStageID.text,
+            'intStageNumber': intStageNumber.text,
+            'strStage': strStage.text,
+            'intCompletionReportID': intCompletionReportID.text,
+            'intMemberStageID': intMemberStageID.text,
+            'dtePassed': convert_date_time(dtePassed),
```

Comment:
As above, you should probably pass `dteExpiration.text` here for consistency with the other fields, and so that your `convert_date_time` actually parses a string, instead of an object.

Reply:
updated it to .text